clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -219,33 +219,37 @@ def drain_node(self, node_name: str, disable_eviction: bool) -> bool:
             return False
 
     def cordon_node(self, node_name: str) -> bool:
+        now = str(arrow.now().timestamp)
         try:
-            self._core_api.patch_node(
-                name=node_name,
-                body={
-                    "spec": {
-                        "unschedulable": True,
-                    },
-                },
+            node = next(filter(lambda x: x.metadata.name == node_name, self._nodes_by_ip.values()))
```

Comment:
This raises an exception if the node is not found, you need to add a default return to `next` to avoid that.

Also, can't we just turn the name into an IP by resolving it, or doing a string transformation, since we use internal DNS names from AWS, and those contain the IP.